Sources/MusicPlayer/Players/SystemMedia.swift

```diff
@@ -164,4 +164,93 @@ private extension Notification.Name {
     static let mediaRemoteNowPlayingApplicationPlaybackStateDidChange = Notification.Name("kMRMediaRemoteNowPlayingApplicationPlaybackStateDidChangeNotification")
 }
 
+#elseif os(Linux)
+
+import playerctl
+
+extension MusicPlayers {
+    
+    public final class FakeSystemMedia: Agent {
```

Comment:
This is closer to `MusicPlayers.NowPlaying`, not `MusicPlayers.SystemMedia` because there isn't a system api on linux. I suggest you subclass `NowPlaying` (`MPRISNowPlaying`?) instead.

If so, you need to modify `MusicPlayers.NowPlaying` so you can change player candidates:
```diff
- public let players: [MusicPlayerProtocol]
+ public var players: [MusicPlayerProtocol] {
+     didSet { selectNewPlayer() }
+ }
```

Reply:
Yes, there isn't a system api on linux, so it's just the `FakeSystemMedia`(or named `MPRISNowPlaying`).
Its behavior is similar to `SystemMedia`. Adding the `SystemMedia` alias makes it possible to obtain the track of current player via a `SystemMedia` instance across platforms.

Using the NowPlaying class is a good idea, but it is marked as final and the players property is immutable, I will modify it and do refactor for `FakeSystemMedia`.